app/modules.go

```diff
@@ -128,3 +132,46 @@ func (SlashingModule) DefaultGenesis(cdc codec.JSONCodec) json.RawMessage {
 
 	return cdc.MustMarshalJSON(genState)
 }
+
+// GenutilModule defines a custom wrapper around the x/genutil module's
+// AppModuleBasic implementation to provide custom genesis state validation.
+type GenutilModule struct {
+	genutil.AppModuleBasic
+}
+
+// ValidateGenesis validates the x/genutil genesis state.
+func (GenutilModule) ValidateGenesis(
+	cdc codec.JSONCodec,
+	encCfg client.TxEncodingConfig,
+	bz json.RawMessage,
+) error {
+
+	var genState genutiltypes.GenesisState
+	if err := cdc.UnmarshalJSON(bz, &genState); err != nil {
+		return fmt.Errorf("failed to unmarshal %s genesis state: %w", genutiltypes.ModuleName, err)
+	}
+
+	txJSONDecoder := encCfg.TxJSONDecoder()
+
+	for i, genTx := range genState.GenTxs {
+		var tx sdk.Tx
+
+		tx, err := txJSONDecoder(genTx)
+		if err != nil {
+			return err
+		}
+
+		msgs := tx.GetMsgs()
+		// if len(msgs) != 1 {
```

Comment:
@facundomedica wdyt...Should we _enforce_ the following?

1. Require exactly 2 messages
2. Require `MsgCreateValidator` and `MsgSetDelegateKeys`